litestar/_openapi/schema_generation/schema.py

```diff
@@ -148,7 +155,19 @@ def _get_type_schema_name(value: Any) -> str:
     Returns:
         A string
     """
-    return cast("str", getattr(value, "__schema_name__", value.__name__))
+
+    if name := getattr(value, "__schema_name__", None):
+        return cast("str", name)
+
+    name = value.__name__
+    if args := getattr(value, "__args__", None):
+        name_parts = [name, "["]
+        name_parts.extend(_get_type_schema_name(a) for a in args)
+        name_parts.append("]")
+
+        return "".join(name_parts)
```

Comment:
Is there a reason to not use `get_args()` here? Also, f-string, and do we need commas to separate the inner names?

```suggestion
    if args := get_args(value):
        inner_parts = ", ".join(_get_type_schema_name(a) for a in args)
        return f"{name}[{inner_parts}]"
```

Reply:
You're absolutely right on both accounts. I should use `get_args` as well as f-strings.